pkg/apis/v1/ec2nodeclass.go

```diff
@@ -453,43 +471,50 @@ func (in *EC2NodeClass) InstanceProfileTags(clusterName string) map[string]strin
 	})
 }
 
+// AMIFamily returns the family for a NodePool based on the following items, in order of precdence:
+//   - ec2nodeclass.spec.amiFamily
+//   - ec2nodeclass.spec.amiSelectorTerms[].alias
+//
+// If an alias is specified, ec2nodeclass.spec.amiFamily must match that alias, or be 'Custom' (enforced via validation).
 func (in *EC2NodeClass) AMIFamily() string {
-	if family, ok := in.Annotations[AnnotationAMIFamilyCompatibility]; ok {
-		return family
+	if in.Spec.AMIFamily != nil {
+		return *in.Spec.AMIFamily
 	}
 	if term, ok := lo.Find(in.Spec.AMISelectorTerms, func(t AMISelectorTerm) bool {
 		return t.Alias != ""
 	}); ok {
-		switch strings.Split(term.Alias, "@")[0] {
-		case "al2":
-			return AMIFamilyAL2
-		case "al2023":
-			return AMIFamilyAL2023
-		case "bottlerocket":
-			return AMIFamilyBottlerocket
-		case "windows2019":
-			return AMIFamilyWindows2019
-		case "windows2022":
-			return AMIFamilyWindows2022
-		}
+		return AMIFamilyFromAlias(term.Alias)
 	}
+	// Unreachable: validation enforces that one of the above conditions must be met
 	return AMIFamilyCustom
 }
 
-func (in *EC2NodeClass) AMIVersion() string {
-	if _, ok := in.Annotations[AnnotationAMIFamilyCompatibility]; ok {
-		return "latest"
+func AMIFamilyFromAlias(alias string) string {
```

Comment:
Ignoring hashing, we need to infer the AMIFamily from the alias for UserData and other bootstrap generation. For hashing, it's just to handle the drift case I described in another comment: where a user updates the `amiFamily` field, but the dynamic `amiFamily` remains the same. 

ex:
```yaml
spec:
  amiSelectorTerms: 
  - alias: al2023@latest
# to
spec:
  amiFamily: al2023
  amiSelectorTerms: 
  - alias: al2023@latest
```
This example shouldn't result in drift, but would if we hashed normally.

Reply:
Got it. I think this makes sense. Just seems like we should document next to the AMIFamily ignore statement that we are intentionally ignoring the hash since we are hashing the AMIFamily value sepearately through a different mechanism